gui.py

```diff
@@ -206,8 +208,11 @@ def start_upload(self):
                 tags='KSP ' + to_utf8(self.tags_tb.text()),
                 token=to_utf8(self.api_token_tb.text()))
         except Exception as e:
-            # QtGui.QMessageBox.critical(self, "Error", '{}'.format(e))
-            QtGui.QMessageBox.critical(self, "Unhandled error", '{}: {}'.format(type(e).__name__, e))
+            io = StringIO.StringIO()
+            traceback.print_exc(file=io)
+            debug_message='{} \n{}'.format(io.getvalue(), self.manager.current_file)
+            QtGui.QMessageBox.critical(self, "Unhandled error [{}]".format(self.manager.current_asset),
+                                       '{}: {}'.format(type(e).__name__,debug_message ))
```

Comment:
the string formatting could break due to 'unexpected' encodings. Are you sure everything is fine there?